components/net/http_loader.rs

```diff
@@ -1256,8 +1256,36 @@ fn http_network_fetch(
         }
     }
 
+    let mut cloned_url = url.clone();
+    let header_strings: Vec<&str> = res
+        .headers()
+        .get_all("Timing-Allow-Origin")
+        .iter()
+        .map(|header_value| header_value.to_str().unwrap())
+        .collect();
+    let wildcard_present = header_strings
+        .iter()
+        .fold(false, |acc, header_str| acc || *header_str == "*");
+    let req_origin_in_timing_allow = header_strings
+        .iter()
+        .map(|header_string| ServoUrl::parse(header_string))
+        .filter(|header_url| header_url.is_ok())
+        .map(|header_url| header_url.unwrap().into_url())
+        .fold(false, |acc, header_url| {
+            acc || header_url.origin() == cloned_url.as_mut_url().origin()
+        });
+
+    if !req_origin_in_timing_allow && !wildcard_present {
+        context
+            .timing
+            .lock()
+            .unwrap()
+            .set_attribute(ResourceAttribute::TimingCheckPassed);
```

Comment:
We should call `mark_timing_check_failed` directly here instead of using set_attribute. That means we can remove the new ResourceAttribute value as well.

Reply:
But if we remove the ResourceAttribute value, then we can't have an early return in set_attribute and we need that early return in case it tries to set it later, right?

Reply:
I'm not sure I understand the issue. `mark_timing_check_failed` will set the `timing_check_passed` member appropriately, and set_attribute can continue to check `timing_check_passed`.

Reply:
Yeah, you're right. I was mistaken.